### ISSUE TITLE
In Equation 8: What do you mean by \hat{M}_{i,j} is the moving average of M_ij 

### ISSUE THREAD

**Issue by jpainam** [hh23333/PVPM#18]
Hi, in Equation 8. 
![image](https://user-images.githubusercontent.com/1701705/95778383-9967b500-0cfa-11eb-9a4b-0e24299d6986.png)

You said, \hat{M} is the moving average of M. Please, what do you mean by that? And how did you implement \hat{M}. I can't really locate that part on your code. Thank you.

**Comment by chenxiangzZ** [comment]
`                if  i!=j and l_g==l_q:
                    cs_ij_=(inputs[i]*inputs[j]).sum(0)   #cross_similarity P*P
                    cs_batch = cs_batch+cs_ij_.detach()
                    cs_ij=torch.diag(cs_ij_)   #cross_similarity P*P
                    s_constr_= (ss_diff_norm[i]*ss_diff_norm[j]).sum(0)
                    ss_batch=ss_batch+s_constr_.detach()
                    W = cs_ij+(s_constr_-self.ss_mean)
                    if use_matching_loss:
                        x_optim, _= self.IQP_solver(W,self.lambd)
                        x_optim = torch.from_numpy(x_optim).cuda(device_id)`
i think the code is here!

**Comment by jpainam** [comment]
Thanks .Ok, i guess, `\hat{M}_{i,j}` is represented as the 
```python
self.register_buffer('ss_mean', torch.zeros((part_num, part_num)))
```
which you update as
```python 
self.ss_mean=self.momentum*self.ss_mean+(1-self.momentum)*ss_batch
```

**Comment by hh23333** [comment]
Yes

**Comment by jpainam** [comment]
Thanks for your reply
